### PR TITLE
[Mono.Android] Remove Mono_UnhandledException lookup

### PR DESCRIPTION
**Pull request by pjcollins**
We can simplify `JNIEnv.PropagateUncaughtException` by removing the
fallback case that would attempt to look for
`System.Diagnostics.Debugger.Mono_UnhandledException`.  In .NET the
`MonodroidRuntime::monodroid_debugger_unhandled_exception` function will
always be defined, and `Mono_UnhandledException` appears to no longer
exist in .NET.  This simplification should also appease our latest set
of API scan results that were complaining about missing documentation
for `Mono_UnhandledException`.

### CODE REVIEW

**Comment by pjcollins** [comment]
I've updated the apkdesc files due to the size reduction in Mono.Android.Runtime.dll:

> stdErr: Error: apkdiff: File 'assemblies/Mono.Android.Runtime.dll' has changed by -430 bytes (-8.35 %). This exceeds the threshold of 5.00 %.

File: src/Mono.Android/Android.Runtime/AndroidRuntimeInternal.cs [hunk 1]
```diff
@@ -6,26 +6,12 @@ namespace Android.Runtime
 {
 	public static class AndroidRuntimeInternal
 	{
-		internal static MethodInfo? mono_unhandled_exception_method = null;
 		internal static Action<Exception> mono_unhandled_exception = RuntimeNativeMethods.monodroid_debugger_unhandled_exception;
```
Comment (jonathanpeppers) [inline src/Mono.Android/Android.Runtime/AndroidRuntimeInternal.cs:9]: Maybe mark this `readonly` while we're at it? Nothing should try to set this `mono_unhandled_exception` field later.

File: src/Mono.Android/Android.Runtime/JNINativeWrapper.cs [hunk 2]
```diff
@@ -73,10 +73,10 @@ public static Delegate CreateDelegate (Delegate dlg)
 			ig.Emit (OpCodes.Leave, label);
 
 			bool  filter = Debugger.IsAttached || !JNIEnvInit.PropagateExceptions;
-			if (filter && AndroidRuntimeInternal.mono_unhandled_exception_method != null) {
+			if (filter && AndroidRuntimeInternal.mono_unhandled_exception.Method != null) {
```
Comment (jonathanpeppers) [inline src/Mono.Android/Android.Runtime/JNINativeWrapper.cs:76]: I don't think `Action<Exception>.Method` can be null? It's not a nullable `MethodInfo?`:

https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Delegate.cs,69

So maybe the only check here is for `filter`?

Comment (pjcollins) [reply src/Mono.Android/Android.Runtime/JNINativeWrapper.cs]: Makes sense thanks, applied

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**